### PR TITLE
Sending emails using gmail

### PR DESCRIPTION
**Pull request by GaelleMarais**
Hi.
@LilyaMel and I have been working on including a way to send emails using gmail. ( cf issue #18 )

 I am using the OpenSSL package (libssl) to create a secure socket, wich is mandatory for communicating with the gmail server:` smtp.gmail.com`

In order to compile the project you need to add `-lssl -crypto -l `to gcc.
Consequently, the full command to compile the project is
`gcc main.c $(pkg-config --libs --cflags libcurl) -lssl -lcrypto -l json-c -std=gnu11`

I added the new command in the `README.md` file and in the `.travis.yml `file.
You also need to install` openssl` (ubuntu) or` libssl `(fedora), command installations are on the `README.md` file.

What we have modified:
  - `analysis.c ,request.c `and `response.c`: add email_score to manage email-related sentences
                 such as 'Can I read my emails?' or 'I want to send an email'.
  -` .travis.yml `: update the compilation command.
  - `README.md :` update the compilation command and add the installation of openssl(fedora)/libssl(ubuntu)
  -` help.txt :` add the email command.
  -` main.c `: include email.c

What's new :
  - `email.c `: Main file to manage email requests.
  - `ssl.c` : A file to create a secure socket layer and communicate with gmail servers.
  - `b64.h `and `b64.c` : A file to encode/decode word in base 64, needed for encode email adresses and passwords.
  - `config_email` : A temporary file where we stock the encoded email adress and password.

Note: We are currently working on a way to read emails and we will make another PR for that.

I hope our work will interest you and that you will accept this PR. If you have any questions I'll respond as soon as possible.
Thank you

### CODE REVIEW

**Comment by ritwik12** [comment]
@GaelleMarais @LilyaMel Hey, this looks really great and shows lots of your hardwork. Thanks to both of you for sending this PR. As this PR is really big so we need some time to review this. Please allow us some time.

**Comment by riya-17** [comment]
Hey @GaelleMarais and @LilyaMel , your PR looks really great, I really appreciate your work. Thanks for this PR.

**Comment by riya-17** [comment]
Hey @ritwik12 I think this PR is good to go

File: README.md [hunk 1]
```diff
@@ -96,16 +96,17 @@ In addition to this, following commands are needed to install the necessary pack
 ```
 $ sudo yum install libcurl libcurl-devel
 $ sudo yum install json-c json-c-devel
+$ sudo dnf install openssl
```
Comment (ritwik12) [inline README.md]: @GaelleMarais openssl-devel is also needed

**Comment by ritwik12** [comment]
```Sending` email ... Error ✗
Please verify your email adress and password.```

Gmail isn't allowing to add an email Id. I am getting above error, Please check

**Comment by ritwik12** [comment]
@GaelleMarais Please check

**Comment by ritwik12** [comment]
@GaelleMarais Code looks good and works fine. Appreciate your hard work for this, Thanks a lot. Please check above issues and errors.

**Comment by GaelleMarais** [comment]
Hello, thank you for your reviews.
I have made the requested changes.

@ritwik12  Have you activated the parameter "Allow access to less secure apps" in your google account ? If it is not activated, only official applications made by Google can access your gmail account. I explained how to do it in the README file. 

It you have activated it and it still doesn't work, I have to know if the problem comes from your gmail account or your device. Can you try again with this account I created especially for testing purposes : 
```
id: mangabountynaia@gmail.com
pwd: MANGABOUNTYNAIA
```

**Comment by ritwik12** [comment]
@GaelleMarais I have tried using the account given by you. It isn't working. Let me recheck.

**Comment by GaelleMarais** [comment]
In file `ssl.c` can you uncomment all lignes marked with DEBUG and show me the error messages?

**Comment by GaelleMarais** [comment]
@ritwik12  Any news ?

**Comment by ritwik12** [comment]
@GaelleMarais Sorry, I got busy. Let me check

**Comment by ritwik12** [comment]
@GaelleMarais Code is  great. Works well. Thanks for contributing such a great feature.

**Comment by ritwik12** [comment]
@GaelleMarais We need one more section in README for Email feature to explain it's working with screenshot just like we have for all other features. Would you like to do that?

**Comment by GaelleMarais** [comment]
Thank you very much. Yes I can do it

**Comment by ritwik12** [comment]
@GaelleMarais Also, feel free to connect and contribute in future. Any help needed or question. Let me know

**Comment by ritwik12** [comment]
@all-contributors please add @GaelleMarais for code

**Comment by ritwik12** [comment]
@all-contributors please add @LilyaMel for code and doc

**Pull request merged**